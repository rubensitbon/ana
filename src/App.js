// @flow
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Routes from './routes';
import { Root } from './components';

type Props = {
  store: Store,
};

const RootComponentWithRoutes = () => (
  <Root>
    <Routes />
  </Root>
);

class App extends React.Component<Props> {
  render() {
    return (
      <Provider store={this.props.store}>
        <Router basename="/">
          <Route path="/" component={RootComponentWithRoutes} />
        </Router>
      </Provider>
    );
  }
}

export default App;
