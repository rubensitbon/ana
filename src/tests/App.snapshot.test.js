// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../App';

describe('<App>', () => {
  it('should render correctly', () => {
    const props = {
      store: {
        subscribe: () => {},
        dispatch: () => {},
        getState: () => {},
      },
    };

    const tree = shallow(<App {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  describe('<Route>', () => {
    it('should render the Root component', () => {
      const props = {
        store: {
          subscribe: () => {},
          dispatch: () => {},
          getState: () => {},
        },
      };

      const tree = shallow(<App {...props} />);
      const route = tree.find('Route');
      expect(route.prop('component')()).toMatchSnapshot();
    });
  });
});
