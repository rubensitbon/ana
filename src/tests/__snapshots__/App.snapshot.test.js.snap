// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> <Route> should render the Root component 1`] = `
<Connect(Root)>
  <routes />
</Connect(Root)>
`;

exports[`<App> should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter
    basename="/"
  >
    <Route
      component={[Function]}
      path="/"
    />
  </BrowserRouter>
</Provider>
`;
